tempodb/encoding/vparquet/wal_block_test.go

```diff
@@ -0,0 +1,193 @@
+package vparquet
+
+import (
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/grafana/tempo/tempodb/backend"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Note: Standard wal block functionality (appending, searching, finding, etc.) is tested with all other wal blocks
+//  in /tempodb/wal/wal_test.go
+
+func TestFullFilename(t *testing.T) {
+	tests := []struct {
+		name     string
+		b        *walBlock
+		expected string
+	}{
+		{
+			name: "basic",
+			b: &walBlock{
+				meta: backend.NewBlockMeta("foo", uuid.MustParse("123e4567-e89b-12d3-a456-426614174000"), VersionString, backend.EncNone, ""),
+				path: "/blerg",
+			},
+			expected: "/blerg/123e4567-e89b-12d3-a456-426614174000+foo+vParquet",
+		},
+		{
+			name: "no path",
+			b: &walBlock{
+				meta: backend.NewBlockMeta("foo", uuid.MustParse("123e4567-e89b-12d3-a456-426614174000"), VersionString, backend.EncNone, ""),
+				path: "",
+			},
+			expected: "123e4567-e89b-12d3-a456-426614174000+foo+vParquet",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := tc.b.walPath()
+			assert.Equal(t, tc.expected, actual)
+		})
+	}
+}
+
+// jpe - restore this when what partial failure conditions exist
```

Comment:
this test was meant to test that partial replays work. may make sense to move into v2?

Reply:
It had already been copied to the v2 folder so good there. Agree this is valuable so attempted a parquet version [here](https://github.com/grafana/tempo/pull/1878/commits/8b0464b9f1e2b66194fd7fed3eb15e34ff2cf3bb).  In this case we can only replay whole flushes, not individual traces like in v2.